heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/WebSink.java

```diff
@@ -192,8 +195,20 @@ public void processRecord(MetricsRecord record) {
     if (isFlatMetrics) {
       metricsCache.putAll(processMetrics(source + "/", record.getMetrics()));
     } else {
-      Map<String, Double> sourceMap = processMetrics("", record.getMetrics());
-      metricsCache.put(source, sourceMap);
+      Map<String, Double> sourceCache;
+      Object sourceObj = metricsCache.getIfPresent(source);
+      if (sourceObj instanceof Map) {
+        @SuppressWarnings("unchecked")
+        Map<String, Double> castObj =  (Map<String, Double>) sourceObj;
+        sourceCache = castObj;
+      } else {
+        sourceCache =  CacheBuilder.newBuilder()
+            .maximumSize(cacheMaxSize)
+            .expireAfterWrite(cacheTtlSec, TimeUnit.SECONDS)
+            .<String, Double>build().asMap();
+      }
+      sourceCache.putAll(processMetrics("", record.getMetrics()));
```

Comment:
I'm not sure what you mean with ignoring the metric type. 

Lets say we have two records `record_1` and `record_2` with this yaml representation:
```
record_1:
  source: source_1
  metrics: 
    metric_1: 1
    metric_2: nop
    metric_3: 3
record_2:
  source: source_1
  metrics:
    metric_1: 2
    metric_4: 4
```
And they are getting send to the WebSink in order.
The WebSink will publish these metrics on a REST endpoint returning json. 
After "record_1" is processed the endpoint will return
```
"source_1": {
    "metric_1": 1.0
    "metric_3": 3.0
}
```
As you can see only numeric metrics are published, "metric_2" is omitted.
After "record_2" is processed the endpoint will return:
```
"source_1": {
  "metric_1": 2.0,
  "metric_3": 3.0,
  "metric_4": 5.0,
}
```
"metric_1" was updated, "metric_3" was not changed, "metric_4" was added by the 2nd record.
The old behavior, before this change, would have returned:
```
"source_1": {
  "metric_1": 2.0,
  "metric_4": 5.0,
}
```

The point of this metrics sink is, that an external process queries periodically (once a min) the 
REST endpoint to collect the metrics. If the two records would get processed right after one another,
there would be the chance that the metrics collector would never see "metric_3".